tests/components/media_player/test_kef.py

```diff
@@ -0,0 +1,234 @@
+
+    def __init__(self, host, port):
+        """Mock for KefSpeaker initializer."""
+        self.__volume = 0.0
+        self.__muted = False
+        self.__source = None
+        self.is_online = True
+        self.host = host
+        self.port = port
+
+    def send_command(self, cmd):
+        """Mock for send command to hardware."""
+        pass
+
+    @property
+    def volume(self):
+        """Mock for getting volume level. None if muted."""
+        if not self.online:
+            raise ConnectionRefusedError("Offline")
+        return self.__volume if not self.__muted else None
+
+    @volume.setter
+    def volume(self, value):
+        """Mock for setting volume level. None to mute."""
+        if not self.online:
+            raise ConnectionRefusedError("Offline")
+        if value:
+            self.__volume = max(0.0, min(1.0, value))
+            self.send_command("volume {}".format(self.__volume))
+        else:
+            self.__muted = True
+            self.send_command("mute")
+
+    @property
+    def source(self):
+        """Mock for getting the input source of the speaker."""
+        if not self.online:
+            raise ConnectionRefusedError("Offline")
+        return self.__source
+
+    @source.setter
+    def source(self, value):
+        """Mock for setting the input source."""
+        if not self.online:
+            raise ConnectionRefusedError("Offline")
+        self.__source = value
+
+    @property
+    def muted(self):
+        """Mock for muted. True if muted, else False."""
+        if not self.online:
+            raise ConnectionRefusedError("Offline")
+        return self.__muted
+
+    @muted.setter
+    def muted(self, value):
+        """Mock for set muted."""
+        if not self.online:
+            raise ConnectionRefusedError("Offline")
+        self.__muted = value
+
+    @property
+    def online(self):
+        """Mock for get if speaker is online or not."""
+        return self.is_online
+
+    # pylint: disable=invalid-name
+    def turnOff(self):
+        """Mock for turn off the speaker."""
+        if not self.online:
+            raise ConnectionRefusedError("Offline")
+        self.is_online = False
+        return True
+
+    # pylint: disable=invalid-name
+    def increaseVolume(self, step=None):
+        """Mock for increase volume."""
+        if not self.online:
+            raise ConnectionRefusedError("Offline")
+        if not self.__muted:
+            self.__volume = self.__volume + (step if step else 0.05)
+
+    # pylint: disable=invalid-name
+    def decreaseVolume(self, step=None):
+        """Mock for decrease volume."""
+        self.increaseVolume(-(step or 0.05))
+
+
+def add_entities_factory(self):
+    """Add devices factory."""
+    def add_entities(devices, update_befor_add=False):
+        """Fake add device."""
+        self.devices += devices
+
+    return add_entities
+
+
+class TestKefMediaPlayer(unittest.TestCase):
```

Comment:
We prefer standalone pytest test functions for new tests.

See the deconz test suite for example.